pkg/container/vector/tools.go

```diff
@@ -15,60 +15,62 @@
 package vector
 
 import (
+	"reflect"
 	"strings"
-
-	"github.com/matrixorigin/matrixone/pkg/common/mpool"
+	"unsafe"
 
 	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"github.com/matrixorigin/matrixone/pkg/common/mpool"
 	"github.com/matrixorigin/matrixone/pkg/container/nulls"
 	"github.com/matrixorigin/matrixone/pkg/container/types"
 	"github.com/matrixorigin/matrixone/pkg/pb/api"
 	"github.com/matrixorigin/matrixone/pkg/pb/plan"
 	"golang.org/x/exp/constraints"
 )
 
-// CheckInsertVector  vector.data will not be nil when insert rows
-func CheckInsertVector(v *Vector, m *mpool.MPool) *Vector {
-	if v.data == nil && v.Typ.Oid != types.T_any && v.Length() > 0 {
-		newVec := New(v.Typ)
-		newVec.isConst = v.isConst
-		val := GetInitConstVal(v.Typ)
-		for i := 0; i < v.Length(); i++ {
-			newVec.Append(val, true, m)
-		}
-		newVec.length = v.length
-		return newVec
-	}
-	return v
-}
-
-func MustTCols[T types.FixedSizeT](v *Vector) []T {
+func MustTCols[T any](v *Vector) []T {
 	// XXX hack.   Sometimes we generate an t_any, for untyped const null.
 	// This should be handled more carefully and gracefully.
 	if v.GetType().Oid == types.T_any {
```

Comment:
is it possible to get rid of t_any?

Reply:
Now plan depends on t_any. May be difficult to remove. Of course, it can be removed. Need to modify some logic of plan.